game-headed/src/main/java/org/triplea/game/client/ui/javafx/screen/ScreenController.java

```diff
@@ -0,0 +1,10 @@
+package org.triplea.game.client.ui.javafx.screen;
+
+/**
+ * A common interface to wrap the logic for swapping screens.
+ *
+ * @param <T> Type of the identifier used to reference different screens.
+ */
+public interface ScreenController<T> {
+  void switchScreen(T identifier);
```

Comment:
Any thoughts to using an enum instead of a generic type? 

We could attach the proper class type to that enum and then would not be coupled to specific class types but instead the enum identifier. Secondly, it'd be very easy to trace down which values and classes are available for `switchScreen` without relying on them inheriting from `Node`. IE: it'll be easier to look at the enum to see all possible instances rather than tracking down everything that extends Node.

Reply:
Interesting idea.
The main reason I used classes for the general case is mostly so that in an IDE it's easy to find the actual implementation.
Using an enum however would make this kinda more reliable during compile time, but I think it requires a little bit more thought because every implementation needs their own enum.
I'd really like to discuss this with you if you think you have a good idea on how to wire up the code using enums.

Reply:
However I'm going to merge this now, so any further changes will be part of another PR

Reply:
As a simple transition idea, maybe long term, an enum holding the class references could be used as a sort of 'shim', a really quick example below:

```

@AllArgsConstructor
enum ScreenNode {
   SETTINGS(SettingsScreen.class),
   MAIN(MainScreen.class);

   @Getter
   private void Class<? extends Node> classReference;
}


class ScreenNavigation {
    void doNavigation(ScreenNode node) {
          Class<? extends Node> nodeClass =  node.getClassReference();
          :
    }
}
```

Thinking about that, it might be even better to use an interface type instead of classes even. That way instead of grabbing a class reference you could invoke the value directly. That would perhaps look something more like this:

```
interface NodeInterface {
    void methodThatDoesStuff();
    Object orMaybeMethodShouldReturnNodeOrSomething();
}


@AllArgsConstructor
enum ScreenNodes {
   SETTINGS(new SettingsNode());

   @Getter
   private final NodeInterface node;
}
```